### PR TITLE
Improve foreign key integrity of dump_domain_data management command

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

https://dimagi-dev.atlassian.net/browse/SAAS-14803

When testing the `dump_domain_data` command with a QA domain, I stumbled into unexpected foreign key violations which led me to see that CaseDeduplicationActionDefinition was not being included in our data dump currently. It seemed best to create a test to ensure we are aware of all foreign key relationships and whether or not they are included in the data dump.
### Safety story

I tested `dump_domain_data` locally to ensure the newly added models work as expected (I assume the query would have failed if it were invalid).

Additionally, this is a tool used by developers only, so it is pretty safe to make changes here.
### Automated test coverage

I added an automated test to ensure we are aware of foreign key relationships in the context of the domain dump moving forward.

### QA Plan

None

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/dump_reload/tests/test_dump_models.py [hunk 1]
```diff
@@ -195,3 +194,34 @@ def _ignore_model(model):
     ]
     assert not uncovered_models, ("Not all Django models are covered by domain dump.\n"
         + '\n'.join(sorted(uncovered_models)))
+
+
+def test_foreign_keys_for_dumped_sql_models_are_also_included():
+    IGNORE_FK_MODELS = [
+        "scheduling.SMSCallbackContent",  # Model is no longer supported, so this would only impact old domains
```
Comment (gherceg) [inline corehq/apps/dump_reload/tests/test_dump_models.py:201]: Instead of ignoring `SMSCallbackContent`, I could add an entry in `APP_LABELS_WITH_FILTER_KWARGS_TO_DUMP` like so:
```
FilteredModelIteratorBuilder('scheduling.SMSCallbackContent', ManyFilters('alertevent__schedule__domain', 'casepropertytimedevent__schedule__domain', 'randomtimedevent__schedule__domain', 'timedevent__schedule__domain')),
```
But it didn't seem worth it since this model is likely not going to be in use anyway.

File: corehq/apps/dump_reload/tests/test_dump_models.py [hunk 2]
```diff
@@ -195,3 +196,24 @@ def _ignore_model(model):
...
+    uncovered_fks_by_model = {}
+    for model in covered_models:
+        foreign_keys = {field.remote_field.model for field in model._meta.fields if isinstance(field, ForeignKey)}
+        uncovered_foreign_keys = foreign_keys - covered_models
+        if uncovered_foreign_keys:
+            uncovered_fks_by_model[get_model_label(model)] = [get_model_label(m) for m in uncovered_foreign_keys]
+
+    assert (
+        not uncovered_fks_by_model
+    ), "Not all foreign key relationships will not be included in the domain dump.\n\n" + "\n\n".join(
```
Comment (kaapstorm) [inline corehq/apps/dump_reload/tests/test_dump_models.py:222]: Nit: I think you mean "Not all ... will be ..."

```suggestion
    ), "Not all foreign key relationships will be included in the domain dump.\n\n" + "\n\n".join(
```

**Review by kaapstorm — APPROVED** [review]

File: corehq/apps/dump_reload/tests/test_dump_models.py (hunk 2 above)
Comment (gherceg) [reply corehq/apps/dump_reload/tests/test_dump_models.py]: I remember seeing this issue, but I clearly did not fix it 😅 thanks for catching!

**Review by millerdev — APPROVED** [review]
The new test is great!

File: corehq/apps/dump_reload/tests/test_dump_models.py (hunk 1 above)
Comment (gherceg) [reply corehq/apps/dump_reload/tests/test_dump_models.py:201]: I'm second guessing ignoring `SMSCallbackContent`. The reasoning is that it seems harder to maintain documentation/awareness around this list so future operators of the `dump_domain_data` command know to confirm the domain does not have any `SMSCallbackContent` objects than it does to just include this filter in the list of dumped models. The command already takes an obscene amount of time to run, so omitting this for that reason isn't sufficient.

TLDR; `users.Permission` is a unique exception that justifies being in `IGNORE_FK_MODELS` where `SMSCallbackContent` really isn't. What do you think @kaapstorm and @millerdev?

Comment (millerdev) [reply corehq/apps/dump_reload/tests/test_dump_models.py:201]: Seems reasonable to include `SMSCallbackContent` if it will not add significant time to the dump.

Comment (gherceg) [reply corehq/apps/dump_reload/tests/test_dump_models.py]: Yep I tested it with a few domains in production and it is very fast. I think I phrased what I attempted to say above a bit better in the commit message: 98361a36494daa3144088ec26036aca3b8647a05.

**Review by millerdev — APPROVED** [review]

**Review by kaapstorm — APPROVED** [review]
